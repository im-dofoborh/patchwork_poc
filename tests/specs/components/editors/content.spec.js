import { mount, createLocalVue } from '@vue/test-utils';
import ContentEditor from '~/components/editors/Content';

describe('ContentEditor component', () => {
  let localVue = null;
  let wrapper = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('content-editor', ContentEditor);
    wrapper = mount(ContentEditor, {
      localVue,
      propsData: {
        path: 'README.md',
      }
    });
  })

  it('renders default variant', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders with content', () => {
    wrapper.setProps({
      content: 'Foo',
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
