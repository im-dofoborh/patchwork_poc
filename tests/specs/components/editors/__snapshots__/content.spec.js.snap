// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentEditor component renders default variant 1`] = `
<div class="sg-content-editor editor-content"><a href="https://github.immediate.co.uk/WCP/patchwork/new/develop/README.md" target="_blank" class="sg-content-editor__link">
    Edit on GitHub
  </a>
  <p>

  </p>
</div>
`;

exports[`ContentEditor component renders with content 1`] = `
<div class="sg-content-editor editor-content"><a href="https://github.immediate.co.uk/WCP/patchwork/edit/develop/README.md" target="_blank" class="sg-content-editor__link">
    Edit on GitHub
  </a>
  <div class="mt-md">Foo</div>
</div>
`;
